usl_models/usl_models/atmo_ml/input_output_sequences.py

```diff
@@ -0,0 +1,117 @@
+            # Standard input sequence construction for day d
+            input_seq = tf.stack(
+                [
+                    inputs[
+                        :, (day - 1) * time_steps_per_day + (time_steps_per_day - 1)
+                    ],  # X_{d-1}^{18}
+                    inputs[:, day * time_steps_per_day + 0],  # X^0
+                    inputs[:, day * time_steps_per_day + 1],  # X^6
+                ],
+                axis=-1,
+            )
+            daily_input_sequences.append(input_seq)
+
+        # Construct input sequences for the current day
+        for t in range(1, time_steps_per_day - 1):
+            input_seq = tf.stack(
+                [
+                    inputs[:, day * time_steps_per_day + (t - 1)],  # X_{t-1}
+                    inputs[:, day * time_steps_per_day + t],  # X^t
+                    inputs[:, day * time_steps_per_day + (t + 1)],  # X^{t+1}
+                ],
+                axis=-1,
+            )
+            daily_input_sequences.append(input_seq)
+
+        # Handle the case for the last sequence in the day
+        if day < num_days - 1:
+            input_seq = tf.stack(
+                [
+                    inputs[
+                        :, day * time_steps_per_day + (time_steps_per_day - 2)
+                    ],  # X_{t-1}
+                    inputs[
+                        :, day * time_steps_per_day + (time_steps_per_day - 1)
+                    ],  # X^t
+                    inputs[:, (day + 1) * time_steps_per_day + 0],  # X_{d+1}^0
+                ],
+                axis=-1,
+            )
+        else:
+            # For the last day, repeat the last available data point
+            input_seq = tf.stack(
+                [
+                    inputs[
+                        :, day * time_steps_per_day + (time_steps_per_day - 2)
+                    ],  # X_{t-1}
+                    inputs[
+                        :, day * time_steps_per_day + (time_steps_per_day - 1)
+                    ],  # X^t
+                    inputs[
+                        :, day * time_steps_per_day + (time_steps_per_day - 1)
+                    ],  # X^t (repeated for no next day)
+                ],
+                axis=-1,
+            )
+        daily_input_sequences.append(input_seq)
+
+        # Construct output sequences for the current day
+        for t in range(0, time_steps_per_day - 1, 2):
+            output_seq = tf.stack(
+                [
+                    labels[:, day * time_steps_per_day + t],  # Y^t
+                    labels[:, day * time_steps_per_day + (t + 1)],  # Y^{t+1}
+                ],
+                axis=-1,
+            )
+            daily_output_sequences.append(output_seq)
+
+        # Instead of stacking, we process sequences directly (e.g., pass to model)
+        for input_seq, output_seq in zip(daily_input_sequences, daily_output_sequences):
+            tf.print("Input sequence:", input_seq.numpy().flatten().tolist())

```

Comment:
Let's guard these print statements behind a `debug: bool` parameter in the function. That way we won't get log spam when not debugging.

Reply:
Done